src/utils/services/mapError.js

```diff
@@ -0,0 +1,28 @@
+const mapError = (state) => {
+  let errorMessage = '';
+  switch (state) {
+    case 'EMAIL_EXISTS':
+      errorMessage = 'The email address is already in use by another account.';
+      break;
+    case 'OPERATION_NOT_ALLOWED':
+      errorMessage = 'Password sign-in is disabled for this project.';
+      break;
+    case 'TOO_MANY_ATTEMPTS_TRY_LATER':
+      errorMessage = 'We have blocked all requests from this device due to unusual activity. Try again later.';
+      break;
+    case 'EMAIL_NOT_FOUND':
+      errorMessage = 'There is no user record corresponding to this identifier. The user may have been deleted.';
+      break;
+    case 'INVALID_PASSWORD':
+      errorMessage = 'The password is invalid or the user does not have a password.';
```

Comment:
Is it possible for a user to not have a password?

Reply:
I took those comments from firebase docus but you're right :)